onnxruntime/core/providers/webnn/builders/impl/resize_op_builder.cc

```diff
@@ -119,9 +119,15 @@ Status ResizeOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder,
   std::vector<int32_t> sizes;
   std::vector<float> scales_hw;
   std::vector<int32_t> sizes_hw;
+  std::vector<int32_t> axes;
+  const bool isNhwc = model_builder.GetPreferredLayout() == DataLayout::NHWC;
   if (input_defs.size() == 3) {  // Use scales.
     ORT_RETURN_IF_NOT(GetResizeScales(initializers, node, scales, logger), "Error getting resize scales");
-    scales_hw = {scales[2], scales[3]};
+    if (isNhwc) {
+      scales_hw = {scales[1], scales[2]};
```

Comment:
Even if the WebNN backend wants a specific layout (say XNNPack NHWC), the ONNX model will still list the scales attribute in the same order. So if model scales = [1,1,3,2], then we wouldn't want to still pass [3,2] to WebNN (not [1,2]), and just reorder the axes?

Reply:
@fdwr, I've verified this, as scales and sizes are all defined as inputs, which will be transposed to preferred layout during initializing.

So currently code is correct.